src/InvariantTest.sol

```diff
@@ -35,6 +47,24 @@ contract InvariantTest is Test {
         excludedSenders_ = _excludedSenders;
     }
 
+    function targetArtifact(string memory newTargetedArtifact_) internal {
+        _targetedArtifacts.push(newTargetedArtifact_);
+    }
+
+    function targetArtifacts() public view returns (string[] memory targetedArtifacts_) {
+        require(_targetedArtifacts.length != uint256(0), "NO_TARGETED_ARTIFACTS");
+        targetedArtifacts_ = _targetedArtifacts;
+    }
+
+    function targetArtifactSelector(FuzzSelector memory newTargetedArtifactSelector_) internal {
+        _targetedArtifactSelectors.push(newTargetedArtifactSelector_);
+    }
+
+    function targetArtifactSelectors() public view returns (FuzzSelector[] memory targetedArtifactSelectors_) {
+        require(targetedArtifactSelectors_.length != uint256(0), "NO_TARGETED_ARTIFACT_SELECTORS");
+        targetedArtifactSelectors_ = _targetedArtifactSelectors;
+    }
```

Comment:
Two UX improvement suggestions, lmk what you think:
1. Re-ordering the contract so the methods user's are intended to call (e.g. `targetContract`) are at the top, and ones forge calls (e.g. `targetContracts`, plural), are at the bottom. Maybe with some comments indicating this difference, to make it easier to learn how to use this file
2. Renaming `targetContract` to `addTargetContract` so it's more clear and descriptive

Reply:
Agree with both of these.

I remember asking about updating the naming conventions here a few months ago. I agree that its confusing that `targetContract` adds a target contract and `targetContracts` returns all targetedContracts.

I'm good with `addTargetContract` naming convention. Will make that change if people agree with it.

Reply:
I wanted to do 1) but hesitated because I didn't know which section label convention to use 😆 I'll use whichever you guys like.

Reply:
> I wanted to do 1) but hesitated because I didn't know which section label convention to use 😆 I'll use whichever you guys like.

Heh, what about something like this:

```solidity
// Functions for users: These are intended to be called in your tests

// Functions for forge: These are called by forge to run invariant tests. You
// do not need to use these or worry about them
```